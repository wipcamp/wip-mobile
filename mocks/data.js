export const newProblem = {
    topic: "Test new problem",
    problem_type_id: 1,
    description: "Test new problem",
    report_id: 1
}

export const problem = [
    {
        id: 1,
        topic: "Test Problem 1",
        description: "Test Description 1",
        problem_type_id: 1,
        report_id: 1,
        is_solve: false,
        not_solve: false,
        created_at: "2018-02-02 17:50:01",
        updated_at: "2018-02-02 17:50:01"
    },
    {
        id: 2,
        topic: "Test Problem 2",
        description: "Test Description 2",
        problem_type_id: 2,
        report_id: 2,
        is_solve: false,
        not_solve: false,
        created_at: "2018-02-02 17:50:02",
        updated_at: "2018-02-02 17:50:02"
    }
]

export const categoryProblem = [
    {
        value: 1,
        label: "Test 1"
    },
    {
        value: 2,
        label: "Test 2"
    }
]

export const filter = 0

export const filterView = [
    {
        value: 0,
        label: "ทั้งหมด"
    },
    {
        label: 1,
        value: "Test 1"
    },
    {
        label: 2,
        value: "Test 2"
    }
]