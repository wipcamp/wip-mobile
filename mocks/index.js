const mockComponent = require.requireActual('react-native/jest/mockComponent')

module.exports = {
    initAll: () => {
      jest.mock('ScrollView', () => {
        const RealScrollView = require.requireActual('ScrollView')
        const ScrollView = mockComponent('ScrollView')
        ScrollView.prototype.render = RealScrollView.prototype.render
        return ScrollView
      })

      jest.mock('Dimensions', () => {
        const Dimensions = {
          get: jest.fn().mockReturnValue({width: 100, height:100})
        }
        return Dimensions
      })
    }
  }