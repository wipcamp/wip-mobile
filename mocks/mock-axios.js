import axios from 'axios'
import mockAxios from 'axios-mock-adapter'

import env from '../config'

const mock = new mockAxios(axios)

export default mock