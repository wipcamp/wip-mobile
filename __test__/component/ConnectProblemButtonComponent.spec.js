/**
 * @jest-environment node
 */

import React from 'react'
import renderer from 'react-test-renderer'

import mockStore from '../../mocks/redux-mock-store'
import { problem } from '../../mocks/data'

import ConnectProblemButton from '../../components/ConnectProblemButtonComponent'

it('rendering', () => {
    const tree = renderer
        .create(
            <ConnectProblemButton
                store={
                    mockStore({
                        ReportReducer: { problem }
                    })
                }
            />
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
})