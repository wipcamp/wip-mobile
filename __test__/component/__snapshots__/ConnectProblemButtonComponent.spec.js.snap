// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 10,
      "flex": 0.4,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 18,
        },
        Object {
          "textAlign": "center",
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
  >
    Reject
  </Text>
</View>
`;
